tests/test_rect2.h

```diff
@@ -0,0 +1,440 @@
+			"clip() with fully enclosed Rect2 should return the expected result.");
+	// The resulting Rect2 is 100 pixels high because the first Rect2 is vertically offset by 100 pixels.
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).clip(Rect2(1200, 700, 100, 100)).is_equal_approx(Rect2(1200, 700, 80, 100)),
+			"clip() with partially enclosed Rect2 should return the expected result.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).clip(Rect2(-4000, -4000, 100, 100)).is_equal_approx(Rect2()),
+			"clip() with non-enclosed Rect2 should return the expected result.");
+}
+
+TEST_CASE("[Rect2] Enclosing") {
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).encloses(Rect2(0, 300, 100, 100)),
+			"clip() with fully contained Rect2 should return the expected result.");
+	CHECK_MESSAGE(
+			!Rect2(0, 100, 1280, 720).encloses(Rect2(1200, 700, 100, 100)),
+			"clip() with partially contained Rect2 should return the expected result.");
+	CHECK_MESSAGE(
+			!Rect2(0, 100, 1280, 720).encloses(Rect2(-4000, -4000, 100, 100)),
+			"clip() with non-contained Rect2 should return the expected result.");
+}
+
+TEST_CASE("[Rect2] Expanding") {
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).expand(Vector2(500, 600)).is_equal_approx(Rect2(0, 100, 1280, 720)),
+			"expand() with contained Vector2 should return the expected result.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).expand(Vector2(0, 0)).is_equal_approx(Rect2(0, 0, 1280, 820)),
+			"expand() with non-contained Vector2 should return the expected result.");
+}
+
+TEST_CASE("[Rect2] Growing") {
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow(100).is_equal_approx(Rect2(-100, 0, 1480, 920)),
+			"grow() with positive value should return the expected Rect2.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow(-100).is_equal_approx(Rect2(100, 200, 1080, 520)),
+			"grow() with negative value should return the expected Rect2.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow(-4000).is_equal_approx(Rect2(4000, 4100, -6720, -7280)),
+			"grow() with large negative value should return the expected Rect2.");
+
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow_individual(100, 200, 300, 400).is_equal_approx(Rect2(-100, -100, 1680, 1320)),
+			"grow_individual() with positive values should return the expected Rect2.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow_individual(-100, 200, 300, -400).is_equal_approx(Rect2(100, -100, 1480, 520)),
+			"grow_individual() with positive and negative values should return the expected Rect2.");
+
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow_margin(MARGIN_TOP, 500).is_equal_approx(Rect2(0, -400, 1280, 1220)),
+			"grow_margin() with positive value should return the expected Rect2.");
+	CHECK_MESSAGE(
+			Rect2(0, 100, 1280, 720).grow_margin(MARGIN_TOP, -500).is_equal_approx(Rect2(0, 600, 1280, 220)),
+			"grow_margin() with negative value should return the expected Rect2.");
+}
+
+TEST_CASE("[Rect2] Has point") {
```

Comment:
We should also test behavior for if the point is on the edge of the Rect2 (we should have at least one test for a negative side and at least one test for a positive side).

Reply:
> Also, why do a lot of the messages say "the expected result" rather than what the result is?

The result is already displayed in the assertion itself, so it's not really important to duplicate it in the message.